Objects/genericaliasobject.c

```diff
@@ -438,19 +438,33 @@ static PyGetSetDef ga_properties[] = {
 static PyObject *
 ga_new(PyTypeObject *type, PyObject *args, PyObject *kwds)
 {
-    if (kwds != NULL && PyDict_GET_SIZE(kwds) != 0) {
-        PyErr_SetString(PyExc_TypeError, "GenericAlias does not support keyword arguments");
+    if (!_PyArg_NoKwnames("GenericAlias", kwds)) {
         return NULL;
     }
-    if (PyTuple_GET_SIZE(args) != 2) {
-        PyErr_SetString(PyExc_TypeError, "GenericAlias expects 2 positional arguments");
+    if (!_PyArg_CheckPositional("GenericAlias", PyTuple_GET_SIZE(args), 2, 2)) {
```

Comment:
I like these specialized error checks, but I doubt the vector call helps much. I'd rather try adding a cache like we have in typing.py.

Reply:
I like the idea of a cache. In term of performance, it's usually hard to beat a cache ;-)

Reply:
@gvanrossum @vstinner cc @serhiy-storchaka 
I agree with guido's opinion, hmm looks like this PR should be closed.
Is it okay to open the sperate PR for specialized error checks?

Reply:
You can open a separated PR changes on the error code.

Yeah, I suggest to close this PR.

Reply:
> You can open a separated PR changes on the error code.
Thanks, I will do.